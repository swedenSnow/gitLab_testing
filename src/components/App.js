import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Login from './Login';
import {
  removeUserFromLocalStorage,
  saveUserToLocalStorage
} from '../utils/localStorage';

class App extends Component {
  static propTypes = {
    user: PropTypes.string.isRequired
  };

  state = {
    user: this.props.user
  };

  loginSuccessful = email => {
    this.setState({ user: email }, () => {
      saveUserToLocalStorage(email);
    });
  };

  logout = () => {
    removeUserFromLocalStorage();
    this.setState({ user: '' });
  };

  render() {
    return (
      <div>
        {this.state.user && (
          <button
            onClick={this.logout}
            className="fixed pin-t p-l bg-purple-dark text-white text-center p-4 rounded-br user"
          >
            {this.state.user}
          </button>
        )}
        <Login loginSuccessful={this.loginSuccessful} />
      </div>
    );
  }
}

export default App;
