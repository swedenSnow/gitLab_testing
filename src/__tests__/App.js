import React from 'react';
import { mount, shallow } from 'enzyme';
import App from '../components/App';

test('should render App corectly', () => {
    const wrapper = shallow(<App user="" />);
    expect(wrapper).toMatchSnapshot();
});

test('should render <App /> without user', () => {
    const wrapper = shallow(<App user="" />);
    expect(wrapper.find('.user').exists()).toBeFalsy();
});

test('should render <App /> with user', () => {
    const user = 'Pelle';
    const wrapper = shallow(<App user={user} />);
    expect(wrapper.find('.user').text()).toEqual(user);
});

test('call the internal method loginSuccessful', () => {
    const user = 'Pelle';
    const wrapper = mount(<App user="" />);
    wrapper.instance().loginSuccessful(user);
    expect(wrapper.state().user).toBe(user);
});

test('call the internal method logout', () => {
    const user = 'Pelle';
    const wrapper = mount(<App user={user} />);
    expect(wrapper.state().user).toBe(user);
    wrapper.instance().logout();
    expect(wrapper.state().user).toBeFalsy();
});
