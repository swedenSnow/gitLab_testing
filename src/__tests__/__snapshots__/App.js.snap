// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App corectly 1`] = `
<div>
  <Login
    loginSuccessful={[Function]}
  />
</div>
`;
